crates/sui-framework/packages/move-stdlib/tests/integer_tests.move

```diff
@@ -104,6 +106,16 @@ public(package) macro fun test_divide_and_round_up($max: _, $cases: vector<_>) {
     })
 }
 
+public(package) macro fun test_mul_div($max: _, $cases: vector<_>) {
+    let max = $max;
+    let cases = $cases;
+    assert_eq!(max.mul_div(max, max.max(1)), max);
+    cases!(max, cases, |_case_pred, case, _case_succ| {
+        assert_eq!(case.mul_div(case, case.max(1)), case); // avoid division by 0
+        // TODO: more test cases?
```

Comment:
What invariants are you able to add to this? I can think of some more for the _ceil case but maybe we can come up with some for mul_div?